### PR TITLE
CORE-183: Break out how to process logging to elastic

### PR DESCRIPTION
**Pull request by ukmadlz**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -57,26 +58,31 @@
     "babel-preset-env": "^1.7.0",
     "chai": "^4.2.0",
     "eslint": "^5.0.0",
-    "eslint-config-airbnb-base": "^13.2.0",
+    "eslint-config-packt-base": "0.0.1",
```
Comment (supachris28) [inline package.json]: carrot

File: src/lib/error-custom.ts [hunk 2]
```diff
@@ -57,28 +64,95 @@ class ErrorCustom extends Error {
...
+   * @param id
+   * @param content
+   */
+  private static async sendToElastic(node: string, id: string, ...content: any): Promise<void> {
+    ErrorCustom.defaultOutput(id, content);
+
+    const date = new Date();
+
+    // Index Name
+    const indexName = process.env.ELASITC_LOGGING_INDEX || `logs-error-custom-${date.getFullYear()}-${date.getMonth()}-${date.getDate()}`;
```
Comment (supachris28) [inline src/lib/error-custom.ts]: Typo in environment variable

File: tests/error-custom.spec.ts [hunk 3]
```diff
@@ -3,9 +3,18 @@
 
 import { expect } from 'chai';
 import sinon from 'sinon';
+import uuid from 'uuid';
 import ErrorCustom from '../src/lib/error-custom';
+import winston from 'winston';
+
+const sandbox = sinon.createSandbox();
+
+const ELASITC_LOGGING_URL = 'http://localhost:9200/';
```
Comment (supachris28) [inline tests/error-custom.spec.ts]: Typo present here too

File: tests/error-custom.spec.ts [hunk 4]
```diff
@@ -88,4 +111,85 @@ describe('Error Custom', () => {
       expect(error.stack).to.be.not.undefined;
     }
   });
+  it('Extend base error with custom logger - ES ENV', () => {
+    sandbox.stub(process, 'env').value({
+      ELASITC_LOGGING_URL,
```
Comment (supachris28) [inline tests/error-custom.spec.ts]: Typo here too

File: tests/error-custom.spec.ts [hunk 5]
```diff
@@ -88,4 +111,85 @@ describe('Error Custom', () => {
...
+        throw new ErrorCustom('It blew up', 500, 101, error);
+      }
+    } catch (error) {
+      expect(error).to.be.instanceof(ErrorCustom);
+      expect(error.message).to.equal('It blew up');
+      expect(error.innerException.message).to.equal('Cannot read property \'toString\' of undefined');
+      expect(error.statusCode).to.equal(500);
+      expect(error.errorCode).to.equal(101);
+      expect(error.manuallyThrown).to.be.true;
+      expect(error.stack).to.be.not.undefined;
```
Comment (supachris28) [inline tests/error-custom.spec.ts]: Need to test that the elastic send is called here and below

File: tests/error-custom.spec.ts [hunk 6]
```diff
@@ -88,4 +111,86 @@ describe('Error Custom', () => {
...
+  });
+  it('Extend base error with custom logger - ES String', () => {
+    try {
+      try {
+        // tslint:disable-next-line: prefer-const
+        let readyToFail;
+        readyToFail.toString();
+        expect('this should not be hit').to.equal(0);
+      } catch (error) {
+        throw new ErrorCustom('It blew up', 500, 101, error, ELASTIC_LOGGING_URL);
```
Comment (supachris28) [inline tests/error-custom.spec.ts]: Use a different value

File: tests/error-custom.spec.ts [hunk 7]
```diff
@@ -88,4 +111,86 @@ describe('Error Custom', () => {
...
+      }
+    } catch (error) {
+      expect(error).to.be.instanceof(ErrorCustom);
+      expect(error.message).to.equal('It blew up');
+      expect(error.innerException.message).to.equal('Cannot read property \'toString\' of undefined');
+      expect(error.statusCode).to.equal(500);
+      expect(error.errorCode).to.equal(101);
+      expect(error.manuallyThrown).to.be.true;
+      expect(error.stack).to.be.not.undefined;
+    }
```
Comment (supachris28) [inline tests/error-custom.spec.ts]: Stub the sendToElastic and check the new value (from above) is passed in

File: tests/error-custom.spec.ts [hunk 8]
```diff
@@ -88,4 +111,86 @@ describe('Error Custom', () => {
...
```
Comment (supachris28) [inline tests/error-custom.spec.ts]: Check elastic isn't called with bad string

**Review by supachris28 — APPROVED** [review]

**Review by jamescodesthings — APPROVED** [review]

**Pull request merged**